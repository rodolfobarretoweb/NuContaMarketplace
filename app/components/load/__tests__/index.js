import React from 'react';
import { StyleSheet } from 'react-native';
import renderer from 'react-native-test-utils';
import checkPropTypes from 'check-prop-types';
import Load from '../index';

describe('Load component', () => {
  it('renders snapshot correctly', () => {
    const load = renderer(<Load />);
    expect(load.toJSON()).toMatchSnapshot();
  });

  it('fails with wrong prop type on `show` prop', () => {
    expect(
      checkPropTypes(Load.propTypes, { show: '' }, 'prop', 'Load')
    ).toContain('Invalid prop `show` of type `string` supplied to `Load`, expected `boolean`.');
  });

  it('fails with wrong prop type on `style` prop', () => {
    expect(
      checkPropTypes(Load.propTypes, { style: { invalidProp: '' } }, 'prop', 'Load')
    ).toMatch('Failed prop type: Invalid props.style key `invalidProp` supplied to `Load`.');
  });

  it('accepts custom style', () => {
    const style = StyleSheet.create({ load: { position: 'absolute' } });
    const load = renderer(
      <Load style={style.load} />
    );
    expect(load.toJSON()).toMatchSnapshot();
  });

  describe('when `show` prop is false', () => {
    it('returns an empty view', () => {
      const load = renderer(<Load show={false} />);
      expect(load.toJSON()).toMatchSnapshot();
    });
  });
});
