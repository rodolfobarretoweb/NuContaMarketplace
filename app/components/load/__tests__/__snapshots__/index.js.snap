// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load component accepts custom style 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      Object {
        "position": "absolute",
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#782698"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      }
    }
  />
</View>
`;

exports[`Load component renders snapshot correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#782698"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      }
    }
  />
</View>
`;

exports[`Load component when \`show\` prop is false returns an empty view 1`] = `<View />`;
